app/component/itinerary/Legs.js

```diff
@@ -188,7 +189,11 @@ export default class Legs extends React.Component {
         !leg.interlineWithPreviousLeg
       ) {
         const mode = getRouteMode({ mode: leg.mode, type: leg.route?.type });
-        legs.push(<TransitLeg mode={mode} {...transitLegProps} />);
+        const feedId = getFeedWithoutId(leg.route?.gtfsId);
+        const changedMode = isExternalFeed(feedId, this.context.config)
+          ? `${mode}-external`
+          : mode;
+        legs.push(<TransitLeg mode={changedMode} {...transitLegProps} />);
       } else if (leg.mode === 'AIRPLANE') {
```

Comment:
See my comment above.

Reply:
I added the logic to getRouteMode. I left the old way in one place, because it was unclear whether the change could be applied